lib/livebook/session/data.ex

```diff
@@ -838,6 +860,51 @@ defmodule Livebook.Session.Data do
     |> set!(notebook: Notebook.update_cell(data.notebook, cell.id, &Map.merge(&1, attrs)))
   end
 
+  defp maybe_invalidate_bound_cells({data, _} = data_actions, cell, prev_cell) do
+    invalidates_bound_cells? =
+      case cell do
+        %Cell.Input{} -> cell.value != prev_cell.value or cell.name != prev_cell.name
```

Comment:
Maybe it is best to have `Cell.Input.changed?` and encapsulate the changed logic in there?

Reply:
Hmm, so then we would need to keep track of `prev_value`, `prev_name`, but we only really need it when updating and is not meaningful out of this context. Accessing the struct before and after update is straightforward here and we already compare `data` with `prev_data` in some places. Also, in this case we need to check for value and name so it's not really `Cell.Input.changed?`, could call it more specifically like `Cell.Input.bindings_invalidated?`, but the bindings logic belongs to data and again this function would be meaningful only in the specific point when we update.

Reply:
You can decide the best name or even to move at all but I was thinking something like `Cell.Input.whatever?(prev_cell)`, yeah.

Reply:
Updated in c0dcc99c33d2467f5c4850cc05076bad87ceda76 :D